src/hotspot/share/runtime/objectMonitor.hpp

```diff
@@ -356,6 +356,9 @@ class ObjectMonitor : public CHeapObj<mtObjectMonitor> {
   void      notify(TRAPS);
   void      notifyAll(TRAPS);
 
+  enum class TryLockResult { Interference = -1, HasOwner = 0, Success = 1 };
+  TryLockResult  TryLock(JavaThread* current);
```

Comment:
Good suggestion! I'll go with the "Another alternative...".

Reply:
Yes, that's a good suggestion.  try_enter() is already exported.  Also change the comment in try_enter():

      // TryLock avoids the CAS and handles deflation.

because it used to be a try_set_owner_from(), and now there's two reasons it can't go back.

Reply:
fixed